samples/python/multicam_image_selector.py

```diff
@@ -0,0 +1,1156 @@
+    plt.close()
+    plt.figure(figsize=(5, 3))
+    if len(all_scores):
+        plt.hist(all_scores, bins=20, alpha=0.4, label="all")
+    if len(sel_scores):
+        plt.hist(sel_scores, bins=20, alpha=0.9, label="selected")
+    plt.xlabel("score")
+    plt.ylabel("count")
+    plt.title(f"{cam}: score distribution")
+    plt.legend(loc="best")
+    plt.tight_layout()
+    plt.savefig(os.path.join(viz_dir, f"{cam}_score_hist.png"))
+    plt.close()
+
+def main(argv: Optional[Sequence[str]] = None) -> None:
+    ap = argparse.ArgumentParser(
+        description="Select best calibration images and emit YAML lists",
+        epilog=(
+            "Examples:\n"
+            "  python3 multicam_image_selector.py --root /data/multicam --out ./yaml --pattern chessboard --rows 7 --cols 10 --per-camera 80 --seed 42 --max-size 1600 --dump-metrics metrics.csv --jobs 8 --viz-out ./viz --cache-file .selector_cache.pkl --resume\n"
+            "  python3 multicam_image_selector.py --root /data --out yaml --pattern aruco --aruco-dict DICT_4X4_1000 --gridboard 7x5 --square 0.03 --separation 0.006 --expected-aruco-markers 20 --selector greedy --greedy-min-dist 0.5 --kmeans-standardize\n"
+            "  python3 multicam_image_selector.py --root /data --out yaml --pattern charuco --rows 4 --cols 11 --aruco-dict DICT_5X5_1000 --square 0.02 --marker 0.014 --require-all-cams --per-camera 60 --fix-orientation --w-sharp 0.5 --w-corners 0.3\n"
+            "  python3 multicam_image_selector.py --root /data --out yaml --pattern circulargrid --rows 4 --cols 11 --circular-type symmetric --per-camera 60"
+        ),
+        formatter_class=argparse.RawDescriptionHelpFormatter,
+    )
+    ap.add_argument("--root", required=True, help="Dataset root containing camera folders")
+    ap.add_argument("--out", required=True, help="Output directory for YAMLs")
+    ap.add_argument(
+        "--pattern",
+        choices=["chessboard", "charuco", "aruco", "circulargrid"],
+        required=True,
+        help="Type of calibration pattern",
+    )
+    ap.add_argument("--rows", type=int, required=True, help="Pattern rows (inner corners or circle rows)")
+    ap.add_argument("--cols", type=int, required=True, help="Pattern cols (inner corners or circle cols)")
+    ap.add_argument("--aruco-dict", default="DICT_5X5_1000", help="Aruco dictionary name for aruco/charuco")
+    ap.add_argument("--circular-type", choices=["symmetric", "asymmetric"], default="symmetric", help="Circle grid type when using circulargrid pattern")
+    ap.add_argument("--square", type=float, default=0.0, help="Square size (meters) for Charuco/GridBoard")
+    ap.add_argument("--marker", type=float, default=0.0, help="Marker size (meters) for Charuco/GridBoard")
+    ap.add_argument("--separation", type=float, default=0.0, help="Marker separation (meters) for GridBoard")
+    ap.add_argument("--gridboard", type=str, help="GridBoard WxH (e.g., 7x5). Only used for --pattern aruco")
```

Comment:
I propose to replace aruco to aruco_grid in  `--pattern`, use --rows and --cols for it and remove --gridboard option.